types/test.ts

```diff
@@ -74,62 +74,62 @@ function testSelectorAsCombiner() {
     subSelector,
   );
 
-  // typings:expect-error
+  // $ExpectError
   selector({foo: ''});
 
-  // typings:expect-error
+  // $ExpectError
   const n: number = selector({bar: {foo: ''}});
 
   const s: string = selector({bar: {foo: ''}});
 }
 
 type Component<P> = (props: P) => any;
 
-declare function connect<S, P, R>(selector: ParametricSelector<S, P, R>):
-  (component: Component<P & R>) => Component<P>;
+declare function connect<S, R, P>(selector: Selector<S, R, P[]>):
+  (component: Component<R & S & ([P] extends [never] ? {} : P)>) => Component<P>;
 
 function testConnect() {
   connect(
     createSelector(
       (state: {foo: string}) => state.foo,
-      foo => ({foo}),
+      foo => ({changeUp: foo}),
```

Comment:
This made the test more obvious that it was "changing" the return type values to something different.